core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/action/ExportDictionaryServlet.java

```diff
@@ -0,0 +1,81 @@
+package be.orbinson.aem.dictionarytranslator.servlets.action;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.servlet.Servlet;
+
+@Component(service = Servlet.class)
+@SlingServletResourceTypes(
+        resourceSuperType = "granite/ui/components/coral/foundation/form",
+        resourceTypes = "aem-dictionary-translator/servlet/action/export-dictionary",
+        methods = "POST")
+public class ExportDictionaryServlet extends SlingAllMethodsServlet {
+    private static final Logger LOG = LoggerFactory.getLogger(ExportDictionaryServlet.class);
+
+    @Override
+    public void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
+        String dictionary = request.getParameter("dictionary");
+        RequestParameter delimiter = request.getRequestParameter("./delimiter");
+
+        response.setContentType("text/csv");
+        response.setHeader("Content-disposition", "attachment; filename=\"dictionary_" + dictionary + ".csv");
+
+        try (PrintWriter writer = response.getWriter()) {
```

Comment:
split into methods for readability and to reduce complexity

Reply:
reformat code for readability